course_UvA-DL/meta-learning/Meta_Learning.py

```diff
@@ -0,0 +1,1350 @@
+    "tensorboards/ProtoMAML/events.out.tfevents.ProtoMAML", "protomaml_fewshot.json",
+    "protomaml_svhn_fewshot.json"
+]
+# Create checkpoint path if it doesn't exist yet
+os.makedirs(CHECKPOINT_PATH, exist_ok=True)
+
+# For each file, check whether it already exists. If not, try downloading it.
+for file_name in pretrained_files:
+    file_path = os.path.join(CHECKPOINT_PATH, file_name)
+    if "/" in file_name:
+        os.makedirs(file_path.rsplit("/", 1)[0], exist_ok=True)
+    if not os.path.isfile(file_path):
+        file_url = base_url + file_name
+        print("Downloading %s..." % file_url)
+        try:
+            urllib.request.urlretrieve(file_url, file_path)
+        except HTTPError as e:
+            print(
+                "Something went wrong. Please try to download the file from the GDrive folder, or contact the author with the full output including the following error:\n",
+                e
+            )
+
+# %% [markdown]
+# ## Few-shot classification
+#
+# We start our implementation by discussing the dataset setup.
+# In this notebook, we will use CIFAR100 which we have already seen in Tutorial 6.
+# CIFAR100 has 100 classes each with 600 images of size $32\times 32$ pixels.
+# Instead of splitting the training, validation and test set over examples, we will split them over classes: we will use 80 classes for training, and 10 for validation and 10 for testing.
+# Our overall goal is to obtain a model that can distinguish between the 10 test classes with seeing very little examples.
+# First, let's load the dataset and visualize some examples.
+
+# %%
+# Loading CIFAR100 dataset
+CIFAR_train_set = CIFAR100(root=DATASET_PATH, train=True, download=True, transform=transforms.ToTensor())
+CIFAR_test_set = CIFAR100(root=DATASET_PATH, train=False, download=True, transform=transforms.ToTensor())
+
+# %%
+# Visualize some examples
+NUM_IMAGES = 12
+CIFAR_images = [CIFAR_train_set[np.random.randint(len(CIFAR_train_set))][0] for idx in range(NUM_IMAGES)]
+CIFAR_images = torch.stack(CIFAR_images, dim=0)
+img_grid = torchvision.utils.make_grid(CIFAR_images, nrow=6, normalize=True, pad_value=0.9)
+img_grid = img_grid.permute(1, 2, 0)
+
+plt.figure(figsize=(8, 8))
+plt.title("Image examples of the CIFAR100 dataset")
+plt.imshow(img_grid)
+plt.axis('off')
+plt.show()
+plt.close()
+
+# %% [markdown]
+# ### Data preprocessing
+#
+# Next, we need to prepare the dataset in the training, validation and test split as mentioned before.
+# The torchvision package gives us the training and test set as two separate dataset objects.
+# The next code cells will merge the original training and test set, and then create the new train-val-test split.
+
+# %%
+# Merging original training and test set
+CIFAR_all_images = np.concatenate([CIFAR_train_set.data, CIFAR_test_set.data], axis=0)
+CIFAR_all_targets = torch.LongTensor(CIFAR_train_set.targets + CIFAR_test_set.targets)
```

Comment:
shall we be more pythonic - no capitals in variable names?
now it seems as kind of hybrid between variable and constant...